pkg/controller/build/reconciler.go

```diff
@@ -1188,12 +1187,35 @@ func (b *buildReconciler) syncMachineOSBuild(ctx context.Context, mosb *mcfgv1.M
 			return fmt.Errorf("could not get MachineConfigPool from MachineOSBuild %q: %w", mosb.Name, err)
 		}
 
-		// An mosb which had previously been forgotten by the queue and is no longer desired by the mcp should not build
 		if mosb.ObjectMeta.Labels[constants.RenderedMachineConfigLabelKey] != mcp.Spec.Configuration.Name {
 			klog.Infof("The MachineOSBuild %q which builds the rendered Machine Config %q is no longer desired by the MCP %q", mosb.Name, mosb.ObjectMeta.Labels[constants.RenderedMachineConfigLabelKey], mosb.ObjectMeta.Labels[constants.TargetMachineConfigPoolLabelKey])
 			return nil
 		}
 
+		oldMCName := mcp.Status.Configuration.Name
+		newMCName := mcp.Spec.Configuration.Name
+
+		oldRendered, err := b.machineConfigLister.Get(oldMCName)
+		if err != nil {
+			return err
+		}
+		newRendered, err := b.machineConfigLister.Get(newMCName)
+		if err != nil {
+			return err
+		}
+
+		old := mcp.DeepCopy()
+		old.Spec.Configuration.Name = mcp.Status.Configuration.Name
+
+		needsImageRebuild, err := b.reconcileImageRebuild(old, mcp)
+		if err != nil {
+			return err
+		}
+		if oldRendered != newRendered && !needsImageRebuild {
+			klog.Infof("MachineOSBuild %q: only layering changes → skipping image build", mosb.Name)
```

Comment:
This log is confusing me a bit, I would think `only layering changes` means a rebuild is actually needed. Maybe we can reword this to make it a bit clearer.